### ISSUE TITLE
Add web-scraping alternative to already retrievable data of YouTube Data API v3

### ISSUE THREAD

**Issue by Benjamin-Loison** [Benjamin-Loison/YouTube-operational-API#91]
As mentioned in #87, being able to retrieve from YouTube operational API YouTube Data API v3 already retrievable data isn't a priority. However in the future it may be interesting, as an alternative to the no-key service for people having exceeded their quota (especially if they have their own instance of the YouTube operational API).

**Comment by Benjamin-Loison** [comment]
As current approach is to have one YouTube UI webpage scraped by YouTube operational API endpoint with a given `part`, maybe merging `playlists` and `playlistItems` makes sense.

**Comment by zakrian07** [comment]
@Benjamin-Loison yes this is me discuss months ago. can we complete this let me know if i can help you to add this
as im using self deploy your source code and i have scrap lot of data and my project is viral i need use proxies and get done this

**Comment by Benjamin-Loison** [comment]
@zakrian07 

You have not answered my question:

> I do not understand your feature request. It seems you want the YouTube operational API to be able to parse https://www.youtube.com/channel/CHANNEL_ID/videos. [YouTube Data API v3 meets your needs](https://stackoverflow.com/a/74579030), no?

**Comment by zakrian07** [comment]
Yes target is to get a channel videos
If you can add vidoes part it would be great as you already scrap clips and
all other tabs info
Scrapping videos is same like 90 percent like clips very minor work will be
for you to add vidoes as part in channel api
Thoughts ?

**Comment by Benjamin-Loison** [comment]
@zakrian07 In my opinion you still have not answered the main part of my question:

> It seems you want the YouTube operational API to be able to parse https://www.youtube.com/channel/CHANNEL_ID/videos. [YouTube Data API v3 meets your needs](https://stackoverflow.com/a/74579030), no?

Hence I am rephrasing it again:

What makes this method: https://stackoverflow.com/a/74579030 using YouTube Data API v3 not solving your problem?

**Comment by zakrian07** [comment]
Yes, it works!
As you know the api have daily limit is 10 k requests
My projects is viral I want prefer scrap methodology same like you did for
clips
I was using your api endpoint sometime it block my ip and it cause downtime
on my app

**Comment by Benjamin-Loison** [comment]
@zakrian07 By *your API endpoint* do you mean [the official instance](https://yt.lemnoslife.com/) no-key endpoint or any web-scraping endpoint? I.e. does the URL limiting you starts with https://yt.lemnoslife.com/noKey/ or not? You can directly share an example of limiting URL to make sure.

Currently I guess that

> your API endpoint sometimes it blocks my IP

refer to [this issue](https://github.com/Benjamin-Loison/YouTube-operational-API/issues/11).

If you are limited by another endpoint than the no-key endpoint, then please try it.

**Comment by zakrian07** [comment]
Thanks for your explanation.
I can share log with you and my ip as well
Really appreciate your help

**Comment by Benjamin-Loison** [comment]
@zakrian07 As you have not answered any of [my questions](https://github.com/Benjamin-Loison/YouTube-operational-API/issues/91#issuecomment-1868271983), share your logs and IP then.

**Comment by zakrian07** [comment]
yes https://yt.lemnoslife.com/noKey/
<https://yt.lemnoslife.com/noKey/videos?part=snippet&id=VIDEO_ID> this
seems a limit quota and it block my ip and  my app get down
this is why
Based on Channels: list
<https://developers.google.com/youtube/v3/docs/channels/list>:
https://yt.lemnoslife.com/channels?part=status,upcomingEvents,shorts,community,channels,about,approval,playlists,snippet,membership&cId=C_ID&id=CHANNEL_ID&handle=HANDLE&forUsername=USERNAME&order=viewCount(&pageToken=PAGE_TOKEN)
I'm wondering if you can add videos in part same as clips. we can use scrap
data without using https://yt.lemnoslife.com/noKey/
<https://yt.lemnoslife.com/noKey/videos?part=snippet&id=VIDEO_ID> or
official  youtube api. it will more convient
let me know if i answer your questions

**Comment by Benjamin-Loison** [comment]
@zakrian07

https://yt.lemnoslife.com/noKey/ is currently powered by `apache2` on a free Oracle Ubuntu having its `iptables` cleared (I say so for possible reproductibility), hence as far as I know there no IP blocking at all. Do you receive any answer from the HTTPS server when you consider being IP blocked? If so, please share it (privately if necessary). (1)

I monitor every minute the correct behavior of the official instance no-key endpoint:
- [here](https://yt0.lemnoslife.com/metrics/checkQuotaLogs.txt) are its logs
- [here](https://yt0.lemnoslife.com/metrics/checkQuota.py) is the monitoring source code

From the logs it does not seem that they were major downtime recently. Could you precise when your app was down the last time? (2)

I clearly understand your request concerning having a web-scraping alternative to the official instance no-key endpoint but from the pieces of information I have on my side I currently do not notice any major issue with the official instance no-key endpoint, so please answer my questions (1) and (2) to let me know if I am wrong.

**Comment by zakrian07** [comment]
![image](https://github.com/Benjamin-Loison/YouTube-operational-API/assets/6196957/80ee0fae-56d2-4b56-8d91-f2144e624258)
this is lat active commit on main branch i have build docker image as well ee7ee425bdd2dca5432d9131a46a2834eb8b1c31

**Comment by zakrian07** [comment]
i have tried to explain as per my knowledge looks like I'm not able to express what I need actually
![image](https://github.com/Benjamin-Loison/YouTube-operational-API/assets/6196957/09353939-03df-447b-bf54-35553b59f3f4)
https://yt.lemnoslife.com/channels?part=about&id=UCeVMnSShP_Iviwkknt83cww&order=viewCount

**Comment by Benjamin-Loison** [comment]
It seems that your issue is about updating your YouTube operational API relying on Docker. I am not keen on that topic but assuming you started your instance with [`docker-compose up`](https://github.com/Benjamin-Loison/YouTube-operational-API/blob/ee7ee425bdd2dca5432d9131a46a2834eb8b1c31/README.md?plain=1#L93), then stopping it with `Ctrl` + `C` and running `docker image rm youtube-operational-api -f`, finally running again `docker-compose up` and *continuing with the new image* (see below) updated my instance on my side, hence theoretically solved your issue:

```
ERROR: The image for the service you're trying to recreate has been removed. If you continue, volume data could be lost. Consider backing up your data before continuing.

Continue with the new image? [yN]y
```

**Comment by zakrian07** [comment]
I will try to delete all volumes and delete image and fresh docker build
Will let you know then

**Comment by zakrian07** [comment]
https://youtubedata.alaskaenterprises.com/channels?part=about&id=UCeVMnSShP_Iviwkknt83cww&order=viewCount
Thank you so much for the correction the issue has been resolved I was just deleting images. but deleting the volume and fresh image pull fixed the issue

**Comment by Benjamin-Loison** [comment]
[[]()](https://discord.com/channels/933841502155706418/933841503103627316/1249913353661579325)

**Comment by Benjamin-Loison** [comment]
https://discord.com/channels/933841502155706418/933841503103627316/1279200163373777027

**Comment by Benjamin-Loison** [comment]
Related to https://discord.com/channels/933841502155706418/933841503103627316/1288291607447212163.

**Comment by Benjamin-Loison** [comment]
https://discord.com/channels/933841502155706418/933841503103627316/1294426983962513570
